feature/feature.go

```diff
@@ -62,6 +62,9 @@ const (
 	// owner: @sedefsavas
 	// alpha: v0.6
 	AutoControllerIdentityCreator featuregate.Feature = "AutoControllerIdentityCreator"
+
+	// IgnitionBootstrapFormat will allow an user to enable alternate machine bootstrap format, viz. Ignition.
+	IgnitionBootstrapFormat featuregate.Feature = "IgnitionBootstrapFormat"
```

Comment:
@surajssd see https://github.com/kubernetes-sigs/cluster-api/pull/4172#discussion_r721154617. We may want to consider renaming the feature gate to `BootstrapFormatIgnition` to allow namespacing along with potential future formats.

Reply:
Will do.

Reply:
Done.